spec/controllers/media_controller_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'rails_helper'
+
+RSpec.describe MediaController, type: :controller do
+  let(:new_mark) { build(:mark) }
+  let(:prez_mark) { build(:mark, :prez) }
+  let(:new_spy) { build(:spy) }
+
+  describe "GET #index" do
+    it "renders the index page" do
+      get :index
+      expect(response).to render_template :index
+    end
+
+    let(:vimeo_params) do
+      {
+        filter: "vimeo"
+      }
+    end
+
+    let(:twitter_params) do
+      {
+        filter: "twitter"
+      }
+    end
+
+    it "renders the index page when the vimeo filter is selected" do
+      new_spy.save
+      session[:spy_id] = new_spy.id
+      get :index, vimeo_params
+      expect(response).to render_template :index
+    end
+
+    it "renders the index page when the twitter filter is selected" do
+      new_spy.save
+      session[:spy_id] = new_spy.id
+      get :index, twitter_params
+      expect(response).to render_template :index
+    end
+  end
+
+  describe "index gets all marks" do
+    it "returns all current spies marks" do
+      new_spy.save
+      session[:spy_id] = new_spy.id
+      get :index
+
+      expect(response).to render_template :index
+    end
+  end
+
+  context "showing the feed" do
+    describe "shows all media" do
+      it "creates new media with vimeo" do
```

Comment:
Code is easier to read if your `describe` and `context` blocks are consistent.
This test file starts off with a describe block describing the controller action `GET #index`, which is a great way to describe tests, but this isn't consistent throughout the test file.